### ISSUE TITLE
Error when sideloading apps (EXC_BAD_ACCESS)

### ISSUE THREAD

**Issue by Smethan** [rileytestut/AltStore#18]
As the title says, whenever I try to sideload an app using the beta, i.e GBA4iOS, Cercube, Rocket for ig, etc., it crashes after around 5 seconds with no communication to AltServer in that time. I can send crash logs, if that would be helpful. And for the device, it's an iPhone XR on iOS 13.1, and all the apps listed above can be sideloaded just fine with impactor

**Comment by eni9889** [comment]
This is due to unsigned framework binaries and there is a pending PR to fix it: https://github.com/rileytestut/AltSign/pull/1

**Comment by Smethan** [comment]
I have tried pulling in your PR and building, now the freshly compiled AltStore just crashes on my phone, says something about a CoreData error

**Comment by eni9889** [comment]
Yea same happens to me, don’t know why the PR would cause that will investigate. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Smethan** [comment]
Seems like its not an issue with your PR, I built it with a clean pull of the repo and I still got the crash, so it must be something else, and it does seem to be isolated to iOS 13, my iOS 12 device worked fine with it...

**Comment by eni9889** [comment]
@Smethan it turns out you need to use Xcode 10.3 to build the project. Xcode 11 has issues.

**Comment by Smethan** [comment]
Good to know. I’ll build there I suppose

**Comment by Smethan** [comment]
Also, any idea how to enable sideloading apps with your own compiled version? As far as it seems, I'm required to download the beta one from within the store

**Comment by Smethan** [comment]
Alrighty, I've compiled it successfully on Xcode 10.3 and ran it on my phone, it works mostly, but I still get the same EXC_BAD_ACCESS error when trying to sign the same apps

**Comment by Smethan** [comment]
I've looked into it and apparently its an issue with dereferencing a null pointer at line 79 in ldid.cpp
`uint32_t data = mach_header.Swap(signature->dataoff);`
The Xcode static analyzer gave the error "Access to field 'dataoff' results in a dereference of a null pointer (loaded from variable 'signature')"

I don't quite know how I would fix this, as I'm not exactly a C++ developer, but maybe it could help someone else out.

**Comment by Smethan** [comment]
Update: I bypassed that error by always making signature non-null, but then I got another similar error, so I don't quite know what the real issue here is. Its looking ldid might just have some issues on iOS 13, or just with A12 devices

**Comment by eni9889** [comment]
Did you use my branch for both altsign and ldid? 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Smethan** [comment]
Yes. I pulled the regular AltStore repo, deleted the altsign sub module, and then checked yours out along with ldid

**Comment by Smethan** [comment]
So, it has been a few days, and it seems to me that the Swap() command seems to be the culprit, as the previous error happened there, and it happens again on the swap command if I bypass that one.
